error/s2n_errno.c

```diff
@@ -94,7 +94,12 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_NON_EMPTY_RENEGOTIATION_INFO, "renegotiation_info should be empty") \
     ERR_ENTRY(S2N_ERR_RECORD_LIMIT, "TLS record limit reached") \
     ERR_ENTRY(S2N_ERR_CERT_UNTRUSTED, "Certificate is untrusted") \
+    ERR_ENTRY(S2N_ERR_CERT_REVOKED, "Certificate has been revoked by the CA") \
+    ERR_ENTRY(S2N_ERR_CERT_EXPIRED, "Certificate has expired") \
     ERR_ENTRY(S2N_ERR_CERT_TYPE_UNSUPPORTED, "Certificate Type is unsupported") \
+    ERR_ENTRY(S2N_ERR_CERT_INVALID, "Certificate is invalid") \
+    ERR_ENTRY(S2N_ERR_CERT_MAX_CHAIN_DEPTH_EXCEEDED, "The maximum certificate chain depth has been exceeded") \
+    ERR_ENTRY(S2N_ERR_CERT_INTERNAL_ERROR, "An internal error has occurred while processing a certificate") \
     ERR_ENTRY(S2N_ERR_INVALID_MAX_FRAG_LEN, "invalid Maximum Fragmentation Length encountered") \
```

Comment:
It doesn't look like S2N_ERR_CERT_INTERNAL_ERROR is used anywhere. If we do have internal errors while processing certs, they should probably use new/existing S2N_ERR_T_INTERNAL errors, not a S2N_ERR_T_PROTO error.

Reply:
I used this a couple of times when there were libcrypto errors:
https://github.com/aws/s2n-tls/blob/dd6b143db08d592ed8485c793845495093d4bbbe/tls/s2n_x509_validator.c#L326-L329
https://github.com/aws/s2n-tls/blob/dd6b143db08d592ed8485c793845495093d4bbbe/tls/s2n_x509_validator.c#L365-L366

I looked at the `S2N_ERR_T_INTERNAL` errors, but couldn't really find a fitting one for these. I made a new internal error called `S2N_ERR_LIBCRYPTO_ERROR` and used this instead. Let me know if you think there's a better error to use, or if I could have made a better new error.

4a47e23edbcdf743402085565b0a0b78db86f086